### DISCUSSION TITLE
ScriptReference/Texture2D.GetPixelBilinear

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#215] [Notes]
# ScriptReference/Texture2D.GetPixelBilinear

https://docs.unity3d.com/ScriptReference/Texture2D.GetPixelBilinear.html

**Comment by mvi** [comment]
At first glance from the code sample alone you may assume that pixel XY to texel UV would be achieved by by normalizing to 0-1 range:  
  

```csharp
float u = x / (texture.width - 1);  
float v = x / (texture.height - 1);
```
  
However as the documentation above says this is not correct, you should instead normalize to one pixel less, as a UV of 1 is would be the same as 0 in repeat filter mode, so instead the conversion should be:  
  

```csharp
float u = x / texture.width;  
float v = x / texture.height;
```
